.github/workflows/build_and_test.yml

```diff
@@ -84,6 +84,34 @@ jobs:
     - name: Run Tests
       run: pushd $KONG_BUILD_TOOLS_LOCATION && make test-kong && popd
 
+  package-and-smoke-test:
```

Comment:
Wondering if this should run for every commit / push / pr; it will considerably add to the overall job time.

Reply:
I could half the difference and only run it on PR jobs?

I'm going to push an empty commit to see what the timing looks like with a primed cache once I've wrapped up the necessary changes to kong-build-tools and docker-kong

Reply:
the job itself takes about 5-7 minutes but is adding negligible time to the PR as we're not over our concurrency limit. It seems like it should be possible to only run this on pull_requests but I couldn't determine the syntax ( https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions )

Reply:
See https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#on

Reply:
Don't we already run the functional tests on tag push, for all packages, though? Does it come as response to some issue we had last release? I'm not 100% confident I see the value of having this job run even on PRs only, but rather on tag time (also, we have daily builds, which also run the smoke tests).

Reply:
We run the functional tests as part of our daily releases as well as part of our proper releases. So rest assured there's zero change we were going to release broken builds.

This comes as a response that we had three PR's with breaking changes that all passed CI. They did fail in the daily builds but that went mostly unnoticed and creates a larger feedback loop than seeing immediately that a PR breaks something

_I can see your points and even provide additional fodder to not add this test_: Possible this is an overreaction. I've known about this blindspot in our CI but we've had almost no bugs / breaking changes that warrented a test such as this. The fact that three happened all in succession might be a statistical anomoly

Reply:
Summarizing for/against

Don't add the tests:
- the bugs wouldn't get released
- pay better attention to daily builds
- seldom have we had bugs this would have caught

Add the tests:
- the breaking change breaks on PR which is arguably the "right place"
- Total added time to PR's is negligible

Reply:
Wasn't aware of the occurrences. In this case, it's a +1 on my side. Could we just make sure it runs only on PR jobs?

Reply:
![image](https://user-images.githubusercontent.com/697188/81838251-e0673980-9513-11ea-8c41-c90a663a7496.png)

Reply:
handled in 93c88f1